### PR TITLE
Added Japanese translation

### PR DESCRIPTION
**Pull request by shelaf**

### CODE REVIEW

**Comment by yanfali** [comment]
@mtei @ai03-2725 any feedback on quality of this translation is appreciated. Many thanks.

**Comment by ai03-2725** [comment]
Will check when I have the time.

File: src/i18n/ja/index.js [hunk 1]
```diff
@@ -1,11 +1,156 @@
...
+      compile: {
+        label: 'Compile',
+        title: 'キーマップのコンパイル'
+      },
+      downloadKeymap: {
+        title: 'keymap.c のみダウンロード',
+        label: 'keymap only'
+      },
+      downloadSource: {
+        title: 'QMK Firmware コードのダウンロード',
```
Comment (ai03-2725) [inline src/i18n/ja/index.js]: Could be ソースコードのダウンロード (Download source code)

File: src/i18n/ja/index.js [hunk 2]
```diff
@@ -1,11 +1,156 @@
...
+      layout: {
+        label: 'layout'
+      },
+      keymapName: {
+        label: 'Keymap Name',
+        placeholder: 'カスタムキーマップ名'
+      },
+      loadDefault: {
+        label: 'Load Default',
+        title: 'QMK Firmware からデフォルトキーマップをロード'
```
Comment (ai03-2725) [inline src/i18n/ja/index.js]: No space needed whenever Japanese follows English
(i.e. QMK Firmware から -> QMK Firmwareから)

**Comment by shelaf** [comment]
Ok, I'll try to fix.

File: src/i18n/ja/index.js [hunk 3]
```diff
@@ -1,11 +1,156 @@
...
+      compile: {
+        label: 'Compile',
+        title: 'キーマップのコンパイル'
+      },
+      downloadKeymap: {
+        title: 'keymap.cのみダウンロード',
+        label: 'keymap only'
+      },
+      downloadSource: {
+        title: 'ソースコードのダウンロード',
```
Comment (mtei) [inline src/i18n/ja/index.js]:
```suggestion
        title: 'QMK全ソースコードのダウンロード',
```
To make the meaning clearer.
上との対比のため、`QMK全` を追加すると分かりやすいのではないでしょうか？

File: src/i18n/ja/index.js [hunk 4]
```diff
@@ -1,11 +1,156 @@
...
+        label: 'Keymap'
+      },
+      downloadToolbox: {
+        label: 'QMK Toolboxを入手'
+      },
+      keycodes: {
+        label: 'Keycodes'
+      },
+      keycodesRef: {
+        label: 'キーコードの参照'
```
Comment (mtei) [inline src/i18n/ja/index.js]:
```suggestion
        label: 'キーコードの説明'
```
or
```suggestion
        label: 'キーコードの説明はこちら'
```
To make the meaning clearer.
ちょっと原文から離れる気もしますが、こちらのほうが分かりやすくないでしょうか？

**Comment by yanfali** [comment]
ども ありがと @mtei

**Comment by yanfali** [comment]
Thank you @ai03-2725 appreciate the input

File: src/i18n/ja/index.js (hunk 2 above)
Comment (shelaf) [reply src/i18n/ja/index.js]: Resolved in f660dcb

File: src/i18n/ja/index.js (hunk 1 above)
Comment (shelaf) [reply src/i18n/ja/index.js]: Resolved in f660dcb

File: src/i18n/ja/index.js [hunk 5]
```diff
@@ -1,11 +1,156 @@
...
+      loadDefault: {
+        label: 'Load Default',
+        title: 'QMK Firmwareからデフォルトキーマップをロード'
+      },
+      compile: {
+        label: 'Compile',
+        title: 'キーマップのコンパイル'
+      },
+      downloadKeymap: {
+        title: 'keymap.cのみダウンロード',
```
[the remark this answers is not in the document]
Comment (shelaf) [reply src/i18n/ja/index.js]: Resolved in a6944e4

File: src/i18n/ja/index.js (hunk 3 above)
Comment (shelaf) [reply src/i18n/ja/index.js]: Resolved in a6944e4

File: src/i18n/ja/index.js (hunk 4 above)
Comment (shelaf) [reply src/i18n/ja/index.js]: Resolved in a6944e4

File: src/store/modules/app/state.js [hunk 6]
```diff
@@ -59,6 +59,7 @@ const state = {
     { value: 'de', label: 'Deutsch' },
     { value: 'fr', label: 'Français' },
     { value: 'es', label: 'Español' },
+    { value: 'ja', label: 'Japanese' },
```
Comment (yanfali) [inline src/store/modules/app/state.js]: Can you please translate this label too please? 
Screen Shot 2019-10-19 at 09 20 01 Something like this? Thank you.

Comment (shelaf) [reply src/store/modules/app/state.js]: This is my intention.
Reason:
* Either "Japanese" or "日本語" is used in multilingual sites and applications.
* "日本語" may be garbled depending on browser font settings (especially for people who don't use Japanese)

**Pull request merged**